### PR TITLE
Combined graphs

### PR DESCRIPTION
**Pull request by rawjazz**
We have coordinated with @frankkwizera and combined the
Time series graph with the Dynamic graph.

### CODE REVIEW

File: seshdash/data/db/influx.py [hunk 1]
```diff
@@ -22,7 +22,12 @@ def __init__(self,database=None):
                                               settings.INFLUX_PASSWORD,
                                               self.db
                                               )
-        #TODO draw this from settings
+#	try:
```
Comment (jlark) [inline seshdash/data/db/influx.py]: why is this commented out?

File: seshdash/data/db/influx.py [hunk 2]
```diff
@@ -161,17 +168,32 @@ def send_object_measurements(self,measurement_dict, timestamp=None, tags={}, dat
...
         return list(self._influx_client.query(query,database=db).get_points())
-
+   
+    
     def create_database(self,name):
         self._influx_client.create_database(name)
 
     def delete_database(self,name):
         self._influx_client.drop_database(name)
+
```
Comment (jlark) [inline seshdash/data/db/influx.py]: Do we need 2 of these functions?

File: seshdash/static/sesh-ui/css/sesh.css [hunk 3]
```diff
@@ -241,6 +244,26 @@ div.awesomplete > ul {
...
 }
+
+/* Time_Series_graph styling */
+
+#update_container{
+  padding-top:60px;
+  margin-bottom:-134px;
+}
+
+#update-butt{
```
Comment (jlark) [inline seshdash/static/sesh-ui/css/sesh.css]: butt means ass :) I would recommend just using the full name button

File: seshdash/static/sesh-ui/js/sesh_dashboard.js [hunk 4]
```diff
@@ -94,52 +213,131 @@ $('#containerhigh').highcharts({
...
+});
 
 
+var csrftoken = getCookie('csrftoken');
+   
+/*-------------------------------END-OF-DEFAULT-GRAPH-------------------------------------*/
 
+ 
+/*----------------------------------------GRAPH-------------------------------------------*/
+  $(".butt").click(function(){
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/sesh_dashboard.js]: Please add some comments in here

File: seshdash/static/sesh-ui/js/sesh_dashboard.js [hunk 5]
```diff
@@ -94,52 +213,131 @@ $('#containerhigh').highcharts({
...
+       drop_choice1 = $("#drop1").val();
+       drop_choice2 = $("#drop2").val();   
+   
+      $.post("/get_measurements_values",{csrfmiddlewaretoken: csrftoken, choice1:drop_choice1, choice2:drop_choice2 , active_site_id:active_site_id},function(data){
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/sesh_dashboard.js]: Some comments also here please

File: seshdash/static/sesh-ui/js/sesh_dashboard.js [hunk 6]
```diff
@@ -250,76 +448,79 @@ get_high_chart( date, HighChartHighPvProduction, HighChartHighCloudCover);
...
-                  for (i=0 ;i < option.length; i++){
-                    /* finding the id of the entered sitename */
-                       if (option[i].value == textinput){
-                             textid = option[i].key;
-                             /* Linking to the asked sitename`s page */
-                             window.location.replace("/dash/" + textid);
-                        }
-                  }
-             { break; }
+/*----------------------------    High chart-------------------------------------*/
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/sesh_dashboard.js]: Please delete

File: seshdash/static/sesh-ui/js/time_series_graph.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                 si_units = data_values.units;
+                 graph_data_values=data_values.graph_values;
+
+
+                  for(var i =0 , l= graph_data_values.length ; i<l ;i++){
+                      graph_data_values[i][0] = graph_data_values[i][0] * 1000;
+                 }
+                  
+            
+         $('#time_series_graph').highcharts({
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/time_series_graph.js]: This time series graph code is repeated too often. We should have a function or a seperate javascript that creates this automatically

File: seshdash/static/sesh-ui/js/time_series_graph.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+            }],
+
+               
+          });
+     });
+});
+
+var measurement_value;
+var time_value;
+$('#update-butt').click(function(){
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/time_series_graph.js]: comments please

File: seshdash/static/sesh-ui/js/time_series_graph.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                 si_units = data_values.units  
+                 graph_data_values = data_values.graph_values
+
+                 for(var i =0 , l= graph_data_values.length ; i<l ;i++){
+                     graph_data_values[i][0] = graph_data_values[i][0] * 1000;
+                 }
+                  
+            
+
+          $('#time_series_graph').highcharts({
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/time_series_graph.js]: Again  this code is repeated too many times. Please ake the highcart creation code more general

File: seshdash/views.py [hunk 10]
```diff
@@ -691,3 +713,87 @@ def historical_data(request):
         context_dict['sort_keys'] = sort_data_dict.keys()
         context_dict['sort_dict'] = sort_data_dict
         return render(request, 'seshdash/historical-data.html', context_dict);
+
+@login_required
+def time_series_graph(request):
```
Comment (jlark) [inline seshdash/views.py]: Comments please.

File: seshdash/views.py [hunk 11]
```diff
@@ -691,3 +713,87 @@ def historical_data(request):
...
+        time_series_values=client.get_measurement_bucket(measurement,time_bucket,'site_name',active_site_name,time_delta)
+
+        graph_values = []
+
+        for values in time_series_values:
+            graph_values.append([values['time'],values['mean']])
+
+        for unformatted_values in graph_values:
+            unformatted_values[0]=get_epoch_from_datetime(datetime.datetime.strptime(unformatted_values[0],"%Y-%m-%dT%H:%M:%SZ"))
+        print graph_values
```
Comment (jlark) [inline seshdash/views.py]: please comment  out or remove print statements. You can use debug logging where appropriate.

File: seshdash/views.py [hunk 12]
```diff
@@ -691,3 +713,87 @@ def historical_data(request):
...
+        context_dict['graph_values']=graph_values
+        context_dict['units']=units  
+        print context_dict     
+        return HttpResponse(json.dumps(context_dict));
+    else:
+        return HttpResponseForbidden()
+    
+@login_required
+def get_measurements_values(request):
```
Comment (jlark) [inline seshdash/views.py]: What's the functional difference between get_measurement_values and time_series_graph/

File: seshdash/views.py [hunk 13]
```diff
@@ -691,3 +713,87 @@ def historical_data(request):
...
+        current_site = active_site_name.site_name
+        SI_unit = BoM_Data_Point.SI_UNITS
+        SI_unit1 = SI_unit[choice_drop_1]
+        SI_unit2 = SI_unit[choice_drop_2]
+        client = Influx()
+        values_drop_1 = client.get_measurement_bucket(choice_drop_1,'10m','site_name',current_site,{'hours': 24})
+        values_drop_2 = client.get_measurement_bucket(choice_drop_2,'10m','site_name',current_site,{'hours': 24})
+        
+        for i in values_drop_1:
+            dropdown1_choice_results.append([i['time'],i['mean']])
```
Comment (jlark) [inline seshdash/views.py]: Can you guarantee this will always be in the right order?

File: settings_local_example.ini [hunk 14]
```diff
@@ -8,10 +8,10 @@ PASSWORD: password
 HOST: hostname
 
 [influx]
-HOST: localhost
+HOST:sesh-dev1.cloudapp.net
```
Comment (jlark) [inline settings_local_example.ini]: Please remove these values and switch to localhost when making a commit

**Comment by jlark** [comment]
Please see comments inline

File: seshdash/views.py [hunk 15]
```diff
@@ -81,6 +89,19 @@ def index(request,site_id=0):
     # Create an object of the get_high_chart_date
     context_dict['high_chart']= get_high_chart_data(request.user,site_id,sites)
     context_dict['site_id'] = site_id
+
+    #Generate measurements in the time_series_graph
+    client=Influx('test_db')
```
Comment (jlark) [inline seshdash/views.py]: This value is hardcoded here
This is a very bad practice. leaving it blank will automatically take the db from the config file!

**Comment by rawjazz** [comment]
On this Branch we have dealt with two graphs 
involving modifying the Daily Data point graph which was configured to use
data generated from the influx db, and creating a Time Series graph which is generated due to the input
or request made by the user including a time range and a measurement name and also on this
data are gained dynamically with the influxdb

File: seshdash/data/db/influx.py [hunk 16]
```diff
@@ -161,61 +171,74 @@ def send_object_measurements(self,measurement_dict, timestamp=None, tags={}, dat
...
 
     def delete_database(self,name):
         self._influx_client.drop_database(name)
 
-    
+    def get_measurements(self,database=None):
+        db = self.db
+        if database:
+           db = database
+        query = "show measurements"
```
Comment (jlark) [inline seshdash/data/db/influx.py]: use capital case for the query please

File: seshdash/views.py [hunk 17]
```diff
@@ -700,3 +720,37 @@ def historical_data(request):
         context_dict['sort_dict'] = sort_data_dict
         return render(request, 'seshdash/historical-data.html', context_dict);
 
+
+#function for Graph Generations   
+@login_required
+def graphs(request):
```
Comment (jlark) [inline seshdash/views.py]: @frank please some comments in this function

File: seshdash/views.py [hunk 18]
```diff
@@ -700,3 +720,37 @@ def historical_data(request):
...
+        date_time = []
+        time = request.POST.get('time','')
+        choices = request.POST.getlist('choice[]')
+        active_id = request.POST.get('active_site_id','')
+        active_site = Sesh_Site.objects.filter(id=active_id)
+        active_site_name = active_site [0]
+        current_site = active_site_name.site_name
+        for choice in choices:
+            data_values = []
+            time_delta_dict = {'24h':{'hours':24},'7d':{'days':7},'30d':{'days':30}}
```
Comment (jlark) [inline seshdash/views.py]: any constant containing dictionaries should be declared in the beginning of the function

File: seshdash/views.py [hunk 19]
```diff
@@ -700,3 +720,37 @@ def historical_data(request):
...
+def graphs(request):
+    if request.method == 'POST':
+        results = {}
+        data_values =[]
+        date_time = []
+        time = request.POST.get('time','')
+        choices = request.POST.getlist('choice[]')
+        active_id = request.POST.get('active_site_id','')
+        active_site = Sesh_Site.objects.filter(id=active_id)
+        active_site_name = active_site [0]
```
Comment (jlark) [inline seshdash/views.py]: You should make sure an active site is returned here before trying to access the [0] element

File: seshdash/views.py [hunk 20]
```diff
@@ -724,16 +724,31 @@ def historical_data(request):
...
+        if active_site != []:
+            active_site_name = active_site [0]
+        else:
+            return HttpResponseBadRequest()
         current_site = active_site_name.site_name
+
+        # processing post request values to be used in the influx queries
         for choice in choices:
             data_values = []
             time_delta_dict = {'24h':{'hours':24},'7d':{'days':7},'30d':{'days':30}}
```
Comment (jlark) [inline seshdash/views.py]: can you also move time delta dict to beginning of the function.
Is there need for it to be declared in every iteration of the `for` loop? @Frankkwizera

File: seshdash/views.py [hunk 21]
```diff
@@ -724,16 +724,31 @@ def historical_data(request):
...
+
+        # Getting values from Post request
         time = request.POST.get('time','')
         choices = request.POST.getlist('choice[]')
         active_id = request.POST.get('active_site_id','')
         active_site = Sesh_Site.objects.filter(id=active_id)
-        active_site_name = active_site [0]
+       
+        # Checking for a valid site_id
+        if active_site != []:
```
Comment (jlark) [inline seshdash/views.py]: in python you can simply say `if active_site` this will return False if the list is empty

**Pull request merged**